src/pattern-library/components/patterns/DialogComponents.js

```diff
@@ -66,6 +74,19 @@ const showDialog = ({ DialogComponent, container, setOpen, props }) => {
 };
 
 export default function DialogComponents() {
+  // Dialog/Modal state for each of the examples
```

Comment:
I concede that the code to handle dialog demos on this pattern-library page is long and wander-y. I considered trying to restructure this and/or extract some into helpers, but felt overwhelmed, so I'll come back to it later.